grafana-plugin/src/containers/MobileAppVerification/MobileAppVerification.tsx

```diff
@@ -138,17 +150,40 @@ const MobileAppVerification = observer(({ userPk }: Props) => {
     </div>
   );
 
+  function clearTimeouts(): void {
```

Comment:
total nitpick; do we have a guideline/decision in terms of `function` vs arrow function? It could be nice to setup an `eslint` rule like [this](https://github.com/TristonJ/eslint-plugin-prefer-arrow) just to keep things consistent.

Reply:
again a nitpick comment, but I would argue in favour of enabling an eslint rule like [`no-use-before-define`](https://eslint.org/docs/latest/rules/no-use-before-define) (this is likely just because I am coming from mostly a Python/clojure background where using a function before it is defined would not compile, but I've also not seen this done very often in JS/TS)

Reply:
Function hoisting is part of the JS environment so they're pretty normal, it's not an anti pattern, but I understand your point. The idea is that if you navigate to a component, 9 out of 10 times you do it to see what's inside the render part, and each time you go to a file you need to scroll to that part. If you move out the helpers (mostly arrow functions from your functional component) after the render part, you'll eliminate this hassle and there's no need for infinite scrolling to find what you're looking for, because most of the times you want to see how props are being passed, or what classes are in there, or whatever DOM structure is there - and that's inside the render. Very rarely you will go to a file to actually change the logic (that's like at most 2-3 out of 10 cases most of the time)

I also find it a bit more cleaner to do it this way, because each function is a separate block, while with arrow functions you just add one after another basically in the same block, but functions would be a bit different because they're hoisted and they make the code look slightly more organized.

Reply:
One other thing we could do is to declare them outside the functional component, but then you'd need to pass in all the required data.